### PR TITLE
demmt/gf100: add compute support

### PR DESCRIPTION
**Pull request by hakzsam**

### CODE REVIEW

File: demmt/object_gf100_compute.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+void decode_gf100_compute_verbose(struct gpu_object *obj, struct pushbuf_decode_state *pstate)
+{
+	int mthd = pstate->mthd;
+	uint32_t data = pstate->mthd_data;
+	struct gf100_compute_data *objdata = obj->class_data;
+
+	if (check_addresses_verbose(pstate, objdata->addresses))
+	{ }
+	else if (mthd == 0x1698) // FLUSH
```
Comment (imirkin) [inline demmt/object_gf100_compute.c]: Why is this needed? That just flushes the on-chip cache afaik... am I wrong?

Comment (hakzsam) [inline demmt/object_gf100_compute.c]: Right, this is actually useless. I forgot to remove that code before the pull request.. :-)

File: demmt/object_gf100_compute.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+		var = varinfo_new(isa_gf100->vardata);
+		varinfo_set_variant(var, "gf100");
+
+		struct region *reg;
+		uint8_t *code = NULL;
+		uint64_t code_addr = objdata->code.address;
+		uint32_t start_id = data;
+		struct gpu_mapping *m = objdata->code.gpu_mapping;
+		if (!m) {
```
Comment (imirkin) [inline demmt/object_gf100_compute.c]: Is this necessary then? I would guess not... would rather leave it off if it's not necessary.

File: demmt/object_gf100_compute.c [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	free(d->addresses);
+	free(d);
+}
+
+void decode_gf100_compute_init(struct gpu_object *obj)
+{
+	struct gf100_compute_data *d = obj->class_data = calloc(1, sizeof(struct gf100_compute_data));
+	obj->class_data_destroy = destroy_gf100_compute_data;
+
+#define SZ 3
```
Comment (imirkin) [inline demmt/object_gf100_compute.c]: Please hook up the texture and image stuff. At least track the image addresses, and print the TSC/TIC's same as gf100_3d does.

File: demmt/object_gf100_compute.c [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+#include "demmt.h"
+#include "log.h"
+#include "nvrm.h"
+#include "object.h"
+
+struct gf100_compute_data
+{
+	struct addr_n_buf code;
+	struct addr_n_buf cb;
+	int data_offset;
```
Comment (imirkin) [inline demmt/object_gf100_compute.c]: How about calling this cb_pos which is what this is.

File: demmt/object_gf100_compute.c (hunk 2 above)
Comment (hakzsam) [inline demmt/object_gf100_compute.c]: I can directly use gpu_mapping_find() here.

File: demmt/object_gf100_compute.c (hunk 3 above)
Comment (hakzsam) [inline demmt/object_gf100_compute.c]: Sure, that was just a preliminary support of compute.

File: demmt/object_gf100_compute.c (hunk 4 above)
Comment (hakzsam) [inline demmt/object_gf100_compute.c]: Yeah.

File: demmt/object_gf100_compute.c (hunk 2 above)
Comment (imirkin) [inline demmt/object_gf100_compute.c]: Do you need to? Can you just use objdata->code.gpu_mapping directly and not worry about the if (!m) case?

Comment (hakzsam) [inline demmt/object_gf100_compute.c]: No, because m is always NULL here. That's why we need to call gpu_mapping_find().

Comment (imirkin) [inline demmt/object_gf100_compute.c]: Ah yeah, looks like they pull the same shit here. Just rewrap my comment since the indent level is less here I guess?

Comment (hakzsam) [inline demmt/object_gf100_compute.c]: Yeah, and it's loosely based on what you did for kepler :-)

File: demmt/object_gf100_compute.c [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+	{
+		struct varinfo *var;
+
+		if (!isa_gf100)
+			isa_gf100 = ed_getisa("gf100");
+
+		var = varinfo_new(isa_gf100->vardata);
+		varinfo_set_variant(var, "gf100");
+
+		// blob has been observed to feed an unmapped address as the
```
Comment (imirkin) [inline demmt/object_gf100_compute.c]: Please keep the logic similar to the other logic... first try the code.mapping. Then if (!m) { ... }. What I was asking you originally was whether the if (!m) could be removed entirely... meaning the whole block, not just doing it unconditionally.

Also while you're at it, make sure to deal with gpu_mapping_find returning NULL.

Comment (hakzsam) [inline demmt/object_gf100_compute.c]: Updated.
